From 9dbd42e1cabf96d189fa1d19e782bb11d4cee21a Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Mon, 24 Oct 2022 04:52:57 +0900
Subject: [PATCH 38/39] kbuild: fix 'Broken pipe' message for AR=gcc-ar

Since commit 321648455061 ("kbuild: use obj-y instead extra-y for
objects placed at the head"), 'make AR=gcc-ar' shows the following
message:

  /usr/bin/ar terminated with signal 13 [Broken pipe]

The resulting vmlinux is correct, but it is better to silence this.

'head -n1' exits after reading the first line, so the pipe is closed.

Use 'sed -n 1p' to eat the stream till the end.

Fixes: 321648455061 ("kbuild: use obj-y instead extra-y for objects placed at the head")
Reported-by: Jiri Slaby <jirislaby@kernel.org>
Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index db7e4055f..6038d284f 100644
--- a/Makefile
+++ b/Makefile
@@ -1147,7 +1147,7 @@ quiet_cmd_ar_vmlinux.a = AR      $@
       cmd_ar_vmlinux.a = \
 	rm -f $@; \
 	$(AR) cDPrST $@ $(KBUILD_VMLINUX_OBJS); \
-	$(AR) mPiT $$($(AR) t $@ | head -n1) $@ $$($(AR) t $@ | grep -F --file=$(srctree)/scripts/head-object-list.txt)
+	$(AR) mPiT $$($(AR) t $@ | sed -n 1p) $@ $$($(AR) t $@ | grep -F --file=$(srctree)/scripts/head-object-list.txt)
 
 targets += vmlinux.a
 vmlinux.a: $(KBUILD_VMLINUX_OBJS) scripts/head-object-list.txt autoksyms_recursive FORCE
-- 
2.38.0.rc1.6.g4fd6c5e444

